File: PTA/Advanced_Level/1006.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int t;
        cin >> t;
    string num, op, cl;
    string num1, num2;
    string early, last;
    for(int i = 0; i < t; ++i) {
        cin >> num >> op >> cl;
        if(!i || op < early) {
            num1 = num;
            early = op;
        }
        if(!i || cl > last) {
            num2 = num;
            last = cl;
        }   
    }
    cout << num1 << ' ' << num2 << endl;
    return 0;
}

File: PTA/Advanced_Level/1006.rs
use std::io::stdin;

fn main() {
    let mut input: String = String::new();
    stdin().read_line(&mut input).unwrap();
    let t: u32 = input.trim().parse::<u32>().unwrap();
    //println!("{}", t);
    let mut num1: String = String::new();
    let mut num2: String = String::new();
    let mut early: String = String::new();
    let mut last: String = String::new();
    let mut flag: bool = true;
    input.clear();
    for _ in 0..t {
        stdin().read_line(&mut input).unwrap();
        let mut v: Vec<String> = input.trim().split(' ').map(|s| s.to_string()).collect();
        if flag || early > v[1] {
            early = v[1].clone();
            num1 = v[0].clone();
        }
        if flag || last < v[2] {
            last = v[2].clone();
            num2 = v[0].clone();
            flag = false;
        }
        input.clear();
    }
    println!("{} {}", num1, num2);
}